[James Kua] iP

James-Kua opened the pull request:
# [Duke]

> 99 little bugs in the code
99 little bugs in the code
Take one down, patch it around
117 little bugs in the code

Duke frees your mind of having to remember things you need to do. It's 

- Fuss-free
- Easy to pick up
- Simple to use

All you need to do is,
1. Download it from [here](https://github.com/James-Kua/ip/releases/tag/A-Jar).
2. _Double-click_ it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] GUI (coming soon)

Here's the `main` method: 
```java
public static void main(String[] args) {
    new Duke(LOCATION).run();
}
```

Commit:
Add Gradle support

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
Level-4

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
A-TextUiTesting

Commit:
“Level-5”

Commit:
Add delete method

Commit:
Add use of Enums

Commit:
Add Save function

Save the tasks in the hard disk automatically whenever the task list changes.
Load the data from the hard disk when Duke starts up.

Commit:
Remove extra data folder

Commit:
Add dates and times

Duke can now understand dates and times.

Commit:
Remove unneccessary imports from Todo class

Commit:
Merge branch 'branch-Level-8'

Commit:
Resolve conflicts during merging

Commit:
Fix loading issues after merge

Commit:
Refactor code as classes

Commit:
Add packages

Divide the classes into suitable packages.

Commit:
Reset to previous commit

Commit:
Add JUnit tests

Commit:
Add JAR file

Commit:
Add A-JavaDoc

Commit:
Add A-CodingStandard

Commit:
Add Level-9

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Add JavaDocs for Duke Class

Commit:
Add JavaDocs for Duke class
q

Commit:
Merge branch 'master' of https://github.com/James-Kua/ip

Commit:
Add Ui printMessage() functionality

Commit:
Make toString modifications to Event class

Commit:
Fix Junit Tests

Commit:
Fix formatSave for Deadline and Event

Commit:
Fix formatSave for Deadline and Event v2.0

Commit:
Ignore duke.txt file

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Commit:
Update mainClassName in build.gradle

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Fix several Checkstyle issues

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Done JavaFX tutorial Part 1

Commit:
Done JavaFX tutorial Part 2

Commit:
Done JavaFX tutorial 3 Iteration 2

Commit:
Done JavaFX tutorial 3 Iteration 2 v2.0

junghon3709 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,212 @@
+                    tasks.deleteTask(input);
+                    break;
+                case TODO:
+                    if (input.substring(5).length() < 1) {
```
This line may be too long

junghon3709 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,212 @@
+import duke.task.*;
+import duke.ui.Ui;
+
+import java.time.LocalDate;
```
I believe java imports should come before all other imports. I'm not sure if it's mentioned explictly in the Java coding standard, but you will run into errors in checkstyle if you do decide to implement it.

junghon3709 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,97 @@
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```
Maybe you could rename `by` into something more descriptive, such as `taskBy`?

junghon3709 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
```
Again, maybe `at` could be more descriptive

junghon3709 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+
+    public abstract String formatSave();
+
+    public String toString() {
```
Mmm, I like `getStatusIcon()`

junghon3709 replied:
Mmm, I like `getStatusIcon()`

junghon3709 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,63 @@
+ * @version: Duke-Level-8
+ */
+public class Ui {
+    private final static String lineBreak = "========================================================================";
```
linebreak is constant, so maybe you should rename it to `LINEBREAK`?

junghon3709 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,63 @@
+     * Prints the logo, together with a greeting.
+     */
+    public void intro() {
+        String logo = " ____        _        \n"
```
Again, maybe `logo` should be `LOGO`, since it is constant.

junghon3709 reviewed (commented):
Very clean code and nice documentation! Just some minor fixes overall :)

Commit:
Done JavaFX tutorial 3 Iteration 2 v3.0, GUI working

Commit:
Done JavaFX tutorial 3 Iteration 3, flip working

Commit:
Done JavaFX tutorial 4

Commit:
Merge branch 'branch-Level-10'

Commit:
Update JavaDocs

Commit:
Restructure packages

Commit:
Improve Duke GUI

Commit:
Improve Duke GUI v2.0

Commit:
Improve Duke GUI v3.0

amzhy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * Responds to user input.
```
Looks great. Just a suggestion: this could be renamed as 'Returns a response... '
In [the coding standard](https://se-education.org/guides/conventions/java/intermediate.html), for method header comments, the first sentence should start in the form `Returns` ..., `Sends` ..., `Adds`. I noticed the same issue in several other places too.

amzhy commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
```
wow, you added Javadoc for `private` methods too

amzhy reviewed (commented):
LGTM👍 . Just a few minor details to fix

James-Kua committed:
Add A-Varargs

James-Kua committed:
Add exit GUI function

James-Kua committed:
Add function to display text in full

James-Kua committed:
Modify exit GUI function

James-Kua committed:
Improve Duke GUI v4.0

James-Kua committed:
Add assertions to Duke and Task classes

James-Kua committed:
Add use of Streams in TaskList

James-Kua committed:
Improve code quality

James-Kua committed:
Merge pull request #1 from James-Kua/branch-A-Streams

Add use of Streams in TaskList

James-Kua committed:
Merge branch 'master' into branch-A-Assertions

James-Kua committed:
Resolve merge conflicts

James-Kua committed:
Merge pull request #2 from James-Kua/branch-A-Assertions

Add assertions to Duke and Task classes

James-Kua committed:
Merge pull request #3 from James-Kua/branch-A-CodeQuality

Improve code quality

James-Kua committed:
Merge branch 'master' into branch-A-CodeQuality

James-Kua committed:
Add CI file

James-Kua committed:
Update GUI to improve text fields

Current text field is too plain

An improved design text field will make the whole GUI more appealing and
nice.

Let's
* Make input font nicer and more readable
* Add colors to the text field and buttons for more visual appeal

James-Kua committed:
Update JavaDocs

James-Kua committed:
Merge pull request #4 from James-Kua/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

James-Kua committed:
Adhere to Checkstyle format

James-Kua committed:
Fix Junit Test (issues with icon)

James-Kua committed:
Ignore ui-test files in gradle.yml

James-Kua committed:
Add C-Help extension

James-Kua committed:
Fix minor issues

James-Kua committed:
Fix DateTime issues

James-Kua committed:
Fix Checkstyle issues

James-Kua committed:
Set theme jekyll-theme-minimal

James-Kua committed:
Update README.md

James-Kua committed:
Update README.md

James-Kua committed:
Fix minor issues to response format

James-Kua committed:
Merge branch 'master' of https://github.com/James-Kua/ip

James-Kua committed:
Add representative screenshot

James-Kua committed:
Update README.md

James-Kua committed:
Add Java compatability to build.gradle

Users encounter problems runnning jar file due to Java version mismatch.

Java 11 is now set as the default Java version during the shadowJar build.

Let's
* Make Java 11 the default Java version during the shadowJar build

James-Kua committed:
Update README.md

James-Kua committed:
Update README.md

James-Kua committed:
Change representative screenshot picture

James-Kua committed:
Merge branch 'master' of https://github.com/James-Kua/ip
